src/plots/cartesian/position_defaults.js

```diff
@@ -72,7 +72,7 @@ module.exports = function handlePositionDefaults(containerIn, containerOut, coer
         // in the axes popover to hide domain for the overlaying axis.
         // perhaps I should make a private version _domain that all axes get???
         var domain = coerce('domain', dfltDomain);
-        if(domain[0] > domain[1] - 0.001) containerOut.domain = dfltDomain;
+        if(domain[0] > domain[1] - (1 / 32768.0)) containerOut.domain = dfltDomain;
```

Comment:
Where does that `32768` number come from?

Reply:
(No need for the trailing `.0` in JS by the way)

Reply:
32768 = 2 ^ 15 is ofcourse the maximum signed short seems to be the maximum width/height on Chrome.
After more research I noticed [this npm module](https://www.npmjs.com/package/canvas-size) updated today with good info! According to that we may reduce that to something like 16,384 for FF. Older versions of IE have lower limits.

Reply:
Interesting. Thanks for the info!

I don't adding a dependency for this is worth it. I suggest finding the max of min values listed. 

Inspecting

https://github.com/jhildenbiddle/canvas-size/blob/52382c95de0b66c447d032f3d8455c6aaaec1fc6/src/index.js#L55-L72

looks like `1 / 4096` will be good enough.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3404/commits/a7c4485f78963c99fc8354b6f29493ee98aa9767

Reply:
Thanks! Maybe you could put a comment above this line explaining where that `4096` comes from.